### PR TITLE
Caracal init

### PR DESCRIPTION
**Pull request by Alex-Welsh**
Includes changes from going through Bobcat and Caracal release notes, plus some general searching for terms like `2023.1` or `NOTE:(upgrade)`

### CODE REVIEW

**Comment by Alex-Welsh** [comment]
Not sure why that docs failure is occurring. @markgoddard It's your magic python config. Any theories?

**Review by seunghun1ee — APPROVED** [review]

**Comment by markgoddard** [comment]
> Not sure why that docs failure is occurring. @markgoddard It's your magic python config. Any theories?

This got me past the error, but I haven't checked the generated URLs.
```
diff --git a/doc/source/conf.py b/doc/source/conf.py
index 052fea34..885d06bf 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -119,9 +119,9 @@ extlinks_projects = {
 }
 
 extlinks = {
-    f"{project}-doc": (f"https://docs.openstack.org/{project}/{current_series}/", "%s documentation")
+    f"{project}-doc": (f"https://docs.openstack.org/{project}/{current_series}/%s", "%s documentation")
     for project in extlinks_projects
 }
-extlinks["skc-doc"] = (f"https://stackhpc-kayobe-config.readthedocs.io/en/stackhpc-{current_series}/", "%s documentation")
-extlinks["kayobe-renos"] = (f"https://docs.openstack.org/releasenotes/kayobe/{current_series}.html", "%s release notes")
-extlinks["kolla-ansible-renos"] = (f"https://docs.openstack.org/releasenotes/kolla-ansible/{current_series}.html", "%s release notes")
+extlinks["skc-doc"] = (f"https://stackhpc-kayobe-config.readthedocs.io/en/stackhpc-{current_series}/%s", "%s documentation")
+extlinks["kayobe-renos"] = (f"https://docs.openstack.org/releasenotes/kayobe/{current_series}.html%s", "%s release notes")
+extlinks["kolla-ansible-renos"] = (f"https://docs.openstack.org/releasenotes/kolla-ansible/{current_series}.html%s", "%s release notes")
```

File: doc/source/operations/upgrading.rst [hunk 1]
```diff
@@ -35,193 +35,70 @@ Notable changes in the |current_release| Release
...
+native support for Prometheus which is now built into HAProxy, metric names may
+have been replaced and/or removed, and in some cases the metric names may have
+remained the same but the labels may have changed. Alerts and dashboards may
+also need to be updated to use the new metrics. Please review any configuration
+that references the old metrics as this is not a backwards compatible change.
+
+Horizon configuration
+---------------------
+The Horizon role has been reworked to the preferred ``local_settings.d``
+configuration model. Files ``local_settings`` and ``custom_local_settings``
```
Comment (markgoddard) [inline doc/source/operations/upgrading.rst:79]: Ouch. I hope someone added a precheck to catch the old files...

Comment (markgoddard) [reply doc/source/operations/upgrading.rst:79]: Nope :(

Comment (markgoddard) [reply doc/source/operations/upgrading.rst:79]: https://review.opendev.org/c/openstack/kolla-ansible/+/923558

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
Docs can be sorted out later.

**Pull request merged**